### ISSUE TITLE
YOLO11 Objects365 pretrained model

### ISSUE THREAD

**Issue by jozefchutka** [ultralytics/ultralytics#22630]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Hi Ultralytics team 

It would be great to have a YOLO11 pretrained model on a large-scale dataset such as Objects365 or Open Images V7 - similar to the existing [yolov5m_Objects365.pt](https://github.com/ultralytics/yolov5/releases/download/v6.0/yolov5m_Objects365.pt) and [yolov8s-oiv7.pt](https://github.com/ultralytics/assets/releases/download/v8.3.0/yolov8s-oiv7.pt).

This is a common request, since training YOLO11 on such datasets is hardly achievable for most users due to hardware limits and dataset size (Objects365 alone exceeds Google Colab storage). Having official pretrained weights would make it much easier to leverage YOLO11’s improved architecture for broad-object detection and transfer learning.

Thanks for considering this, it would be a huge help to the community

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @jozefchutka, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started quickly—an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

For feature requests like YOLO11 pretrained weights on large-scale datasets (e.g., Objects365/Open Images V7) ❇️:
-Please share any additional context on intended use cases, desired model sizes (e.g., n/s/m/l/x), and expected categories/benchmarks.  
-If you have public references or results from community experiments using YOLO11 on these datasets, adding brief summaries and logs can help the team evaluate feasibility and prioritization.  
-You can also react with 👍 on this issue to help us gauge community interest.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You can use YOLOE. It's pretrained on Objects365.

https://docs.ultralytics.com/models/yoloe/

**Comment by jozefchutka** [comment]
Thanks for the heads up. I decided to use [yoloe-11s-seg-pf.pt](https://github.com/ultralytics/assets/releases/download/v8.3.0/yoloe-11s-seg-pf.pt) which seems to use 4585 classes ([RAM TAG](https://raw.githubusercontent.com/xinyu1205/recognize-anything/main/ram/data/ram_tag_list.txt)?) which is a bit more than Objects365 but is suitable.

I would still love to see regular detection models pretrained on Objects365. In the meanhile there is a pretrained nano model available [YOLOv11n + Objects365](https://huggingface.co/NRtred/yolo11n_object365/blob/main/yolo11n_object365.pt)

**Comment by glenn-jocher** [comment]
Thanks, noted—we don’t currently publish YOLO11 detection checkpoints pretrained on Objects365/Open Images, but we’ll keep this enhancement open; meanwhile you can convert a YOLOE Seg checkpoint to a detection-only checkpoint that runs like YOLO11 Detect:

```python
from ultralytics import YOLOE
from ultralytics.utils.patches import torch_load

det = YOLOE("yoloe-11s.yaml")
state = torch_load("yoloe-11s-seg.pt")  # or use yoloe-11s-seg-pf.pt
det.load(state["model"])
det.save("yoloe-11s-seg-det.pt")
```

If you decide to self-train, the dataset recipe is here: see the Objects365 dataset guide and its YAML at [Objects365 dataset docs](https://docs.ultralytics.com/datasets/detect/objects365/) and [Objects365.yaml](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/datasets/Objects365.yaml); Open Images V7 training is outlined at the [Open Images V7 dataset guide](https://docs.ultralytics.com/datasets/detect/open-images-v7/). If you or others have reproducible training logs and evals for community O365 weights, we’d welcome a PR to add them to the assets.

**Comment by jozefchutka** [comment]
Thank @glenn-jocher , I tried your convertion steps:

```py
det = YOLOE("yoloe-11s.yaml")
state = torch_load("yoloe-11s-seg.pt")  # or use yoloe-11s-seg-pf.pt
det.load(state["model"])
det.save("yoloe-11s-seg-det.pt")
```

...and exported to .onnx, which eventually, in inference time (onnxruntime-web), returned `NaN` instead of reasonable values for confidence. I tried both seg.pt and seg-pf.pt with the same results.

I ended up using the following:

```py
model = YOLOE("yoloe-11s-seg.pt")
names = ["Person", "Sneakers", ...other 395 Objects365 classes]
model.set_classes(names, model.get_text_pe(names))
model.export(format="onnx", opset=17, simplify=True, half=True, device=0)
```

Which exported working .onnx and seem to detect just the 365 provided classes. I am not sure this is the optimal way though...

**Comment by glenn-jocher** [comment]
Good approach—exporting after `set_classes()` is the recommended path for web; to avoid NaNs in onnxruntime-web prefer FP32 and disable graph NMS, then run NMS client‑side:

```python
from ultralytics import YOLOE

model = YOLOE("yoloe-11s-seg.pt")
names = [...]  # your 365-class list
model.set_classes(names, model.get_text_pe(names))
model.export(format="onnx", opset=17, simplify=True, dynamic=False, half=False, nms=False)
```

If you’d like us to investigate the NaNs from the `*-seg-det.pt` export, please share a minimal repro (export args, onnxruntime-web version, full logs) and the output of `yolo checks`; export options are listed in the [Export documentation](https://docs.ultralytics.com/tasks/detect/#export). We'll keep this enhancement open for native YOLO11 Objects365 weights.

**Comment by jozefchutka** [comment]
Hi @glenn-jocher , thank you for looking into this, here is the reproduction steps for `NaN` issue:

Generate model:

```py
from ultralytics import YOLOE
from ultralytics.utils.patches import torch_load

det = YOLOE("yoloe-11s.yaml")
state = torch_load("yoloe-11s-seg.pt")  # or use yoloe-11s-seg-pf.pt
det.load(state["model"])
det.save("yoloe-11s-seg-det.pt")

model = YOLOE("yoloe-11s-seg-det.pt")
model.export(format="onnx", opset=17, simplify=True, dynamic=False, nms=False, device=0, half=True)
```

Run following test.html browser:

```html
<script src="https://cdn.jsdelivr.net/npm/onnxruntime-web@1.22.0/dist/ort.webgpu.min.js"></script>
<script>
(async () => {
const session = await ort.InferenceSession.create("yoloe-11s-seg-det.onnx", {executionProviders: ["webgpu"]});
const inputData = new Float16Array(1 * 3 * 640 * 640);
const inputTensor = new ort.Tensor("float16", inputData, [1, 3, 640, 640]);
const results = await session.run({images:inputTensor});
console.log(results.output0.data.slice(40000, 40010))
})()
</script>
```

This will print a slice of data into console like:

```
Float16Array(10) [NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, buffer: ArrayBuffer(20), byteLength: 20, byteOffset: 0, length: 10, Symbol(Symbol.toStringTag): 'Float16Array']
```

- tested with `wasm` instead of `webgpu` - same results.
- tested with FP32 version which does not return NaN, but the values are super low basically returning no detections on real image

**Comment by glenn-jocher** [comment]
Thanks for the very clear repro, that’s super helpful. The fact that FP16 NaNs only appear on the seg→det converted checkpoint (while your direct `yoloe-11s-seg.pt` export works) points to a precision issue in that converted graph, possibly compounded by simplification. As a next step, could you try exporting the `yoloe-11s-seg-det.pt` model in FP32 with simplification disabled and test it in your ORT‑Web setup using a `Float32Array` input:

```python
from ultralytics import YOLOE

model = YOLOE("yoloe-11s-seg-det.pt")
model.export(format="onnx", opset=17, simplify=False, dynamic=False, nms=False, half=False)
```

If this still produces NaNs or unusably small scores, please attach the exported `.onnx` (zipped) and paste the output of `yolo checks` so we can dig into it on our side; ONNX‑specific tips and caveats for Ultralytics models are collected in the [ONNX export guide](https://docs.ultralytics.com/integrations/onnx/).

**Comment by jozefchutka** [comment]
Github fails to attach zipped .onnx (32MB) but you can run my steps in [google colab with T4 GPU](https://colab.research.google.com/drive/1Dz9j15eDPItLoDaA85B2wNW2z_wcVg92) to generate the exact same:

```sh
!curl https://github.com/ultralytics/assets/releases/download/v8.3.0/yoloe-11s-seg.pt --output yoloe-11s-seg.pt -L
!pip install ultralytics
from ultralytics import YOLOE
from ultralytics.utils.patches import torch_load

det = YOLOE("yoloe-11s.yaml")
state = torch_load("yoloe-11s-seg.pt")  # or use yoloe-11s-seg-pf.pt
det.load(state["model"])
det.save("yoloe-11s-seg-det.pt")

model = YOLOE("yoloe-11s-seg-det.pt")
model.export(format="onnx", opset=17, simplify=False, dynamic=False, nms=False, device=0, half=False)
```

This one (FP32, no simplify) does not return NaN-s but it still does not detect anything and following dims it only recognizes 80 classes (COCO?). Happy to help you digging further, but is this anyhow helpful to my case as I am looking for something in size of Objects365?

**Comment by glenn-jocher** [comment]
You’re right: the `yoloe-11s.yaml` seg→det path produces a COCO‑style 80‑class head, so even in FP32 it won’t give you a 365‑class Objects365 detector and isn’t really helpful for your use case. For now, your `set_classes(...)` + ONNX export approach is the right way to get an Objects365‑scale vocabulary from YOLOE. If you’re up for it, please open a small separate issue summarizing the seg→det ONNX behaviour (80‑class head + FP16 NaNs) so we can track that bug, and we’ll keep this thread focused on the pretrained‑weights request.

**Comment by jozefchutka** [comment]
Following the advice from https://github.com/ultralytics/ultralytics/issues/22744#issuecomment-3563600976 , these steps exports .onnx model with Objects365 classes:

```py
model = YOLOE("yoloe-11s.yaml").load("yoloe-11s-seg.pt")
names = [...]  # your 365-class list
model.set_classes(names)
model.export(format="onnx", opset=17, simplify=True, dynamic=False, nms=False, device=0, half=True)
```

which is slightly smaller and faster while maintaining same detections compared to:

```py
model = YOLOE("yoloe-11s-seg.pt")
names = [...]  # your 365-class list
model.set_classes(names, model.get_text_pe(names))
model.export(format="onnx", opset=17, simplify=True, dynamic=False, nms=False, device=0, half=True)
```

**Comment by glenn-jocher** [comment]
That makes sense and matches what we’d expect: instantiating `YOLOE("yoloe-11s.yaml").load("yoloe-11s-seg.pt")` and then calling `set_classes(names)` keeps YOLOE in a closed‑set configuration and lets the exporter strip unused open‑vocab components, so the ONNX ends up smaller/faster while preserving the same 365‑class behaviour—this is a good pattern for your Objects365 use case, and if you ever see FP16 issues you can keep the same flow and just switch `half=False` on export.
